[Samsung][docs] Update to the new template

SS-JIA opened the pull request:
Summary:
Title says it all!

Add docs for the Samsung backend based on the template introduced in https://github.com/pytorch/executorch/pull/14873.

digantdesai reviewed (approved):
Thanks @SS-JIA. Let's get it reviewed by someone at Samsung? Stamping to unblock merging if you have to.

SS-JIA committed:
[Samsung][docs] Update to the new template

Summary:
Title says it all!

Add docs for the Samsung backend based on the template introduced in https://github.com/pytorch/executorch/pull/14873.
